controller/iteration_blackbox_test.go

```diff
@@ -1080,3 +1080,183 @@ func (rest *TestIterationREST) TestIterationDelete() {
+			delete(allChildren, i.ID)
+		}
+		require.Empty(t, allChildren)
+
+		// when
+		children, err = rest.db.Iterations().LoadChildren(svc.Context, itr3.ID)
+		// then
+		require.Nil(t, err)
+		require.Len(t, children, 1)
+
+		allChildren = map[uuid.UUID]struct{}{
+			itr4.ID: {},
+		}
+		for _, i := range children {
+			delete(allChildren, i.ID)
+		}
+		require.Empty(t, allChildren)
+	})
+
+	rest.T().Run("update fail - parent of root iteraton", func(t *testing.T) {
+		fxt := tf.NewTestFixture(t, rest.DB, tf.CreateWorkItemEnvironment(), tf.Iterations(2))
+		rootItr := fxt.Iterations[0]
+		newParentIDStr := fxt.Iterations[1].ID.String()
+		payload := minimumUpdatePayloadWithParent()
+		payload.Data.Relationships.Parent.Data.ID = &newParentIDStr
+		payload.Data.ID = &rootItr.ID
+		svc, ctrl := rest.SecuredControllerWithIdentity(fxt.Identities[0])
+		test.UpdateIterationForbidden(t, svc.Context, svc, ctrl, rootItr.ID.String(), &payload)
+	})
+
+	rest.T().Run("update fail - non-existing parent of iteraton", func(t *testing.T) {
+		fxt := tf.NewTestFixture(t, rest.DB, tf.CreateWorkItemEnvironment(), tf.Iterations(2))
+		itr1 := fxt.Iterations[1]
+		newParentIDStr := "73048351-a4c0-4cf6-aff9-1fa3d7576ac0"
+		payload := minimumUpdatePayloadWithParent()
+		payload.Data.Relationships.Parent.Data.ID = &newParentIDStr
+		payload.Data.ID = &itr1.ID
+		svc, ctrl := rest.SecuredControllerWithIdentity(fxt.Identities[0])
+		test.UpdateIterationNotFound(t, svc.Context, svc, ctrl, itr1.ID.String(), &payload)
+	})
+
+	rest.T().Run("update fail - invalid UUID parent of iteraton", func(t *testing.T) {
+		fxt := tf.NewTestFixture(t, rest.DB, tf.CreateWorkItemEnvironment(), tf.Iterations(2))
+		itr1 := fxt.Iterations[1]
+		newParentIDStr := "/"
+		payload := minimumUpdatePayloadWithParent()
+		payload.Data.Relationships.Parent.Data.ID = &newParentIDStr
+		payload.Data.ID = &itr1.ID
+		svc, ctrl := rest.SecuredControllerWithIdentity(fxt.Identities[0])
+		test.UpdateIterationBadRequest(t, svc.Context, svc, ctrl, itr1.ID.String(), &payload)
+	})
+
+	rest.T().Run("update fail - parent UUID is same as subject iteraton", func(t *testing.T) {
+		fxt := tf.NewTestFixture(t, rest.DB, tf.CreateWorkItemEnvironment(), tf.Iterations(1))
+		itr := fxt.Iterations[0]
+		newParentIDStr := itr.ID.String()
+		payload := minimumUpdatePayloadWithParent()
+		payload.Data.Relationships.Parent.Data.ID = &newParentIDStr
+		payload.Data.ID = &itr.ID
+		svc, ctrl := rest.SecuredControllerWithIdentity(fxt.Identities[0])
+		test.UpdateIterationForbidden(t, svc.Context, svc, ctrl, itr.ID.String(), &payload)
+	})
+
+	rest.T().Run("update fail - valid parent but from different space", func(t *testing.T) {
+		fxt1 := tf.NewTestFixture(t, rest.DB, tf.Iterations(1, tf.SetIterationNames("alpha")))
+		fxt2 := tf.NewTestFixture(t, rest.DB, tf.Iterations(1, tf.SetIterationNames("beta")))
+		alpha := fxt1.IterationByName("alpha")
+		beta := fxt2.IterationByName("beta")
+		newParentIDStr := alpha.ID.String()
```

Comment:
Can you just write: `fxt1.IterationByName("alpha").ID.String()`? There's no need to persist `alpha` as a variable because it is only used once.

Reply:
done https://github.com/fabric8-services/fabric8-wit/pull/1770/commits/3ccaca03459814b927d7260a2a1896f96bf65ac5